plugins/backend/wiz-backend/src/service/WizClient.ts

```diff
@@ -0,0 +1,191 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from 'node-fetch';
+import { Config } from '@backstage/config';
+
+export class WizClient {
+  private clientId: string;
+  private clientSecret: string;
+  private tokenUrl: string;
+  private wizAPIUrl: string;
+  private accessToken: string | null = null;
+  private refreshToken: string | null = null;
+  private tokenExpiresAt: number | null = null;
+
+  constructor(config: Config) {
+    this.clientId = config.getString('wiz.clientId');
+    this.clientSecret = config.getString('wiz.clientSecret');
+    this.tokenUrl = config.getString('wiz.tokenUrl');
+    this.wizAPIUrl = config.getString('wiz.wizAPIUrl');
+  }
+
+  async fetchAccessToken() {
+    const params = new URLSearchParams();
+    params.append('client_id', this.clientId);
+    params.append('client_secret', this.clientSecret);
+    params.append('grant_type', 'client_credentials');
+    params.append('audience', 'wiz-api');
+
+    const response = await fetch(this.tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: params,
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch access token: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+    this.accessToken = data.access_token;
+    this.refreshToken = data.refresh_token;
+    this.tokenExpiresAt = Date.now() + data.expires_in * 1000;
+  }
+
+  async refreshAccessToken() {
+    if (!this.refreshToken) {
+      throw new Error('Refresh token not available');
+    }
+
+    const params = new URLSearchParams();
+    params.append('client_id', this.clientId);
+    params.append('client_secret', this.clientSecret);
+    params.append('refresh_token', this.refreshToken);
+    params.append('grant_type', 'refresh_token');
+    params.append('audience', 'wiz-api');
+
+    const response = await fetch(this.tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: params,
+    });
+
+    if (!response.ok) {
+      throw new Error(`Failed to refresh access token: ${response.statusText}`);
+    }
+
+    const data = await response.json();
+    this.accessToken = data.access_token;
+    this.refreshToken = data.refresh_token;
+    this.tokenExpiresAt = Date.now() + data.expires_in * 1000;
+  }
+
+  async ensureValidToken() {
+    if (
+      !this.accessToken ||
+      (this.tokenExpiresAt && Date.now() >= this.tokenExpiresAt)
+    ) {
+      await this.refreshAccessToken();
+    }
+  }
+
+  async getIssuesForProject(projectId: string) {
```

Comment:
It would be nice if this function returned had a return type. But not a massive thing.